includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -573,8 +608,7 @@ private function get_payment_method_from_request() {
 		$payment_method = ! empty( $_POST['wcpay-payment-method'] ) ? wc_clean( $_POST['wcpay-payment-method'] ) : null; //phpcs:ignore WordPress.Security.ValidatedSanitizedInput.MissingUnslash
```

Comment:
cc @luizreis I think he might have thoughts on this after he added the saved cards functionality?

Reply:
Yeah, that seems like it could cause some bugs on that edge case. After our session today, I feel like the way we're handling `tokens` vs `payment_methods` in the `Payment_Information` class is better and could also be applied here.

The only remaining usage of this method is in the `create_setup_intent` function, so I went ahead and removed it in favor of the `Payment_Information` class in https://github.com/Automattic/woocommerce-payments/pull/826/commits/c9f7e301e41baacf899ed15daadb09a341eaa207. What do you guys think?

PS: I'm sorry for adding more stuff to be reviewed here, we can fix this in another PR if you'd like.

Reply:
I like the way you did it. I'd say this is exactly one of the benefits of having the wrapper class 🙂. I'll mark this as resolved!